core/filesystem.schema.go

```diff
@@ -79,6 +81,9 @@ type Filesystem {
 
 	"push a filesystem as an image to a registry"
 	pushImage(ref: String!): Boolean!
+
+	"write a file at path"
+	writeFile(path: String!, contents: String!, permissions: String = "0664" ): Filesystem
```

Comment:
I'm setting `permissions` as string since graphql doesn't handle octales natively like Go or CUE. An alternative here instead could be to create a new scalar called `FileMode` that translates to the same go type. Not sure what's going to happen with the graphiql schema validation if we try to set something like 0600 or 0o600 as a default argument value since. 

Thoughts?

Reply:
We've spoken about this with @aluzzardi in the retreat. We think it's best if we leave it as a string. 

cc @sipsma for one last :eyes: before merging :pray:

Reply:
Yeah I think having an octal `FileMode` is a good idea in the long term but can be saved for a followup after the api refactor and such. It's probably one of those things that's easier said than done, so no need right now.